src/parse/mod.rs

```diff
@@ -488,4 +490,39 @@ pub(crate) mod tests {
         let buf: String = buf.into_iter().collect();
         assert!(parse(&buf).is_none());
     }
+
+    prop_compose! {
+        fn arb_vec_result_locatable_token()(tokens in any::<Vec<Token>>()) -> Vec<CompileResult<Locatable<Token>>> {
+            tokens.into_iter().map(|token| Ok(Locatable { data: token, location: Location::default()})).collect()
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn proptest_peek_equals_token(
+            first in any::<Token>(),
+            tokens in arb_vec_result_locatable_token()
+            ) {
+            let mut parser = Parser::new(Locatable { data: first, location: Location::default() }, tokens.into_iter(), false);
```

Comment:
It would be nice for this to be a `prop!` impl eventually but we don't need it for a first pass.